docs/basic_construction.md

```diff
@@ -0,0 +1,155 @@
+      $$
+        d \leftarrow \Bigg\lceil \frac{2 u \lambda_{rel}}{\log e}\Bigg\rceil.
+      $$
+  - The probability that a tuple of full size is selected $q$ is inversely proportional to $d$, approximating $\lambda_{rel} / d$. 
+    - Since $d$ is linked to $u$, $q$ is approximately inversely proportional to $u$.
+      $$
+        q \leftarrow \frac{2 \lambda_{rel}}{d \log e}.
+      $$
+    - A smaller $q$ reduces the chance of accepting invalid proofs, enhancing security, but also decreases the probability of finding a valid proof, necessitating a larger $d$.
+
+### Prover Algorithm
+The prover's objective is to convince the verifier that they know a large subset of elements $S_p$, with size exceeding the threshold $n_f$, without revealing the entire set or any specific subset of size greater than $n_f$.
+Instead, the prover constructs a compact proof $(t, s_1, ..., s_u)$ that satisfies the following:
+- The proof demonstrates that the prover possesses knowledge of a set $S_p$ satisfying a predicate $R$, with the assurance that $|S_p| > n_f$, where $n_f$ is a threshold strictly smaller than $n_p$ and $|S_p| \geq n_p$.
+- The prover outputs only a small sequence of $u$ elements from $S_p$, rather than transmitting the full set, ensuring communication efficiency.
+- By using random oracles $H_1$ and $H_2$, the prover ensures that:
+  - The proof is hard to forge by an adversary who knows at most $n_f$ elements from $S_p$.
+  - A prover who knows a subset $S_p$ with size greater than $n_p$ can construct a valid proof with high probability, leveraging the filtering and validation mechanisms of $H_1$ and $H_2$.
+- The constructed tuple $(t, s_1, ..., s_u)$ satisfies:
+  - $H_1(t, s_1, ..., s_i) = 1$ for all prefixes up to length $u$,
+  - $H_2(t, s_1, ..., s_u) = 1$ for the final tuple.
+
+In summary, the prover's task is to use recursive construction via DFS to find and output a valid tuple that demonstrates knowledge of $S_p$ without directly revealing the full set.
+
+### Verifier Algorithm
+The verifier's objective is to validate the prover's claim by ensuring that the submitted proof $(t, s_1, ..., s_u)$ satisfies the conditions of the ALBA protocol. Specifically:
+- Verify that the proof tuple satisfies the random oracles' checks for both $H_1$ and $H_2$.
+ - Check that the proof is well formed, as specified in the protocol.
+- The verifier ensures that the proof was generated by a prover possessing a large subset $S_p$. If the prover knows fewer than $n_f$ elements, the probability of passing verification should be negligible.
+- The verifier performs efficient checks proportional to the proof size $u$, making the process scalable.
+
+### Functions
+#### Proving
+---
+> $\mathsf{Prove}(S_p) \rightarrow \pi$
+> - Input:
+>   - $S_p:~~$ `prover_set`, a set of elements to be proven.
+> - Output:
+>   - $\pi:~~$ `proof`, a valid proof $(t, s_1, \ldots, s_u)$ or $\emptyset$.
```

Comment:
Here and below we should probably use the "bottom" symbol instead of the empty set. `\bot`. It's usually used to denote errors.